pkg/sliexporter/vshnminio_controller/vshnminio_controller.go

```diff
@@ -0,0 +1,158 @@
+)
+
+type VSHNMinioReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+
+	ProbeManager       probeManager
+	StartupGracePeriod time.Duration
+	MinioDialer        func(service, name, namespace, organization, sla, endpointURL string, ha bool, opts minio.Options) (*probes.VSHNMinio, error)
+}
+
+type probeManager interface {
+	StartProbe(p probes.Prober)
+	StopProbe(p probes.ProbeInfo)
+}
+
+//+kubebuilder:rbac:groups=vshn.appcat.vshn.io,resources=xvshnminios,verbs=get;list;watch
+//+kubebuilder:rbac:groups=vshn.appcat.vshn.io,resources=xvshnminios/status,verbs=get
+//+kubebuilder:rbac:groups=vshn.appcat.vshn.io,resources=vshnminios,verbs=get;list;watch
+//+kubebuilder:rbac:groups=vshn.appcat.vshn.io,resources=vshnminios/status,verbs=get
+
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch
+
+func (r *VSHNMinioReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
+	l := log.FromContext(ctx).WithValues("namespace", req.Namespace, "instance", req.Name)
+	l.Info("Reconciling VSHNRedis")
+
+	inst := &vshnv1.XVSHNMinio{}
+	err = r.Get(ctx, req.NamespacedName, inst)
+
+	if apierrors.IsNotFound(err) || inst.DeletionTimestamp != nil {
+		l.Info("Stopping Probe")
+		r.ProbeManager.StopProbe(probes.ProbeInfo{
+			Service:   vshnMinioServiceKey,
+			Name:      req.Name,
+			Namespace: req.Namespace,
+		})
+		return ctrl.Result{}, nil
+	}
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if inst.Spec.WriteConnectionSecretToRef.Name == "" {
+		l.Info("No connection secret requested. Skipping.")
+		return ctrl.Result{}, nil
+	}
+
+	probe, err := r.getMinioProber(ctx, inst)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return ctrl.Result{}, err
+	}
+
+	if apierrors.IsNotFound(err) {
+		l.WithValues("credentials", inst.Spec.WriteConnectionSecretToRef.Name, "error", err.Error()).
+			Info("Failed to find credentials. Backing off")
+		res.Requeue = true
+		res.RequeueAfter = 30 * time.Second
+
+		if time.Since(inst.GetCreationTimestamp().Time) < r.StartupGracePeriod {
+			// Instance is starting up. Postpone probing until ready.
+			return res, nil
+		}
+	}
+
+	l.Info("Starting Probe")
+
+	r.ProbeManager.StartProbe(probe)
+	return res, nil
+
+}
+
+func (r VSHNMinioReconciler) getMinioProber(ctx context.Context, inst *vshnv1.XVSHNMinio) (prober probes.Prober, err error) {
+	l := log.FromContext(ctx).WithValues("namespace", inst.ObjectMeta.Labels[claimNamespaceLabel], "instance", inst.ObjectMeta.Labels[claimNameLabel])
+
+	creds := corev1.Secret{}
+
+	fmt.Println("looking for secret: ", SLIBucketName, " in namespace: ", "vshn-minio-"+inst.Name)
+
+	err = r.Get(ctx, types.NamespacedName{
+		Name:      SLIBucketName,
+		Namespace: "vshn-minio-" + inst.Name,
+	}, &creds)
+
+	if err != nil {
+		return nil, err
+	}
+
+	org := inst.GetLabels()[utils.OrgLabelName]
+	if org == "" {
```

Comment:
This will always result in `vshn` org on any APPUiO Manage cluster, as they don't have this label.

If the label is empty:
Either we need some way to determine what organization any given APPUiO Managed cluster belongs to, or we make it configurable via some flag or env var. Then we can just set it in SYN during deployment.

Probably would make sense to have that logic for the other probers as well.

Reply:
I talked wtih Sebastian about that and we have 2 additional labels coming from each managed Openshift Cluster with syn-tenant-id and cluster name. Isn't that sufficient?

Reply:
Yeah that would work, but in that case we need:
* not setting the org to vshn if it's empty
* teach the sla reporter to look at `syn-tenant-id` if the organization label is not set

We should open a followup for that.

Reply:
https://github.com/vshn/appcat/pull/95/files#diff-3243738ee01fda0e17f64917b3b41de603fa1685b108a6fc6d89e052b1470ec8R129

removed defaulting logic